posts/2021-03-19-microprofile-4-21003.adoc

```diff
@@ -0,0 +1,151 @@
+
+[#MP4]
+=== MicroProfile 4.0
+
+Microprofile 4.0 is a major version update complete with updates to the following 8 MicroProfile components:
+
+* MicroProfile Config 2.0
+
+* MicroProfile Fault Tolerance 3.0
+
+* MicroProfile Health 3.0
+
+* MicroProfile JWT Authentication 1.2
+
+* MicroProfile Metrics 3.0
+
+* MicroProfile OpenAPI 2.0
+
+* MicroProfile OpenTracing 2.0
+
+* MicroProfile Rest Client 2.0
+
+As this is such a large update standalone blogs have been written for both MicroProfile 4.0 and a few select components, a deep-dive into MicroProfile 4.0 can be found in the blog linked below with further component-specific blogs due to release over the coming weeks:
+
+* link:https://openliberty.io/blog/2021/03/19/microprofile40-open-liberty-21003.html[MicroProfile 4.0 support in Open Liberty 21.0.0.3]
+
+To make use of MicroProfile 4.0 you must add the feature to both `pom.xml` and `server.xml`:
+
+Add the following to your `server.xml`:
+[source, xml]
+----
+<server>
+  <featureManager>
+    <feature>microProfile-4.0</feature>
+  </featureManager>
+</server>
+----
+
+And add the following to your `pom.xml`:
+[source, xml]
+----
+<dependency>
+    <groupId>org.eclipse.microprofile</groupId>
+    <artifactId>microprofile</artifactId>
+    <version>4.0.1</version>
+    <type>pom</type>
+    <scope>provided</scope>
+</dependency>
+----
+
+//end::features[]
+
+[#bugs]
+== Significant bugs fixed in this release
+
+We’ve spent some time fixing bugs. The following sections describe just some of the issues resolved in this release. If you’re interested, here’s the link:https://github.com/OpenLiberty/open-liberty/issues?q=label%3Arelease%3A21003+label%3A%22release+bug%22[full list of bugs fixed in 21.0.0.3].
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/15927[Fix the injection of Optional Lists and Sets (MP Config)]
++
+Previously in Open Liberty it was not possible to inject an `Optional<List<String>>` or `Optional<Set<String>>` using `@ConfigProperty`. This was problematic as in the newly release MP Config 2.0 removed the ability to inject an empty list as an empty value, as it will treated the same as a missing value. This would lead to diffuclties when writing backwards-compatible code for both MP Config 2.0 and 1.x versions. This issue has now been fixed and there should be no issues injecting optional lists, for more information view the link:https://openliberty.io/docs/latest/reference/feature/mpConfig-2.0.html[MP Config 2.0 documentation].
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/12215[Read jvm.options file from the intended directory]
++
+According to the link:[Open Liberty documentation] the `jvm.options` file is inteded to be read from the `${wlp.user.dir}/shared/jvm.options` directory to allow for custom JVM options to be loaded. Prior to Open Liberty 21.0.0.3 this file was erroneously fetched from the `${WLP_INSTALL_DIR}/usr/shared/jvm.options` path, this behaviour has now been corrected.
```

Comment:
inteded should be intended (spelling mistake)

Reply:
Thanks for catching.